src/app/merge-diff/merge-diff-container/merge-diff-container.component.ts

```diff
@@ -53,8 +57,9 @@ export class MergeDiffContainerComponent implements OnInit {
     }
 
     // const sourceId = this.uiRouterGlobals.params.sourceId;
-    // const targetId = this.uiRouterGlobals.params.targetId;
+   //  const targetId = this.uiRouterGlobals.params.targetId;
     this.domainType = this.uiRouterGlobals.params.catalogueDomainType;
+    this.source = this.mergeService.source;
```

Comment:
Is a data model passed into the `MergeDiffContainerComponent` via the service to avoid further backend calls? If so, not sure how I feel about this method of passing the input, seems like it relies too much on knowing how to setup the component/route to the page before being able to use it.

I think `UIRouter` is able to pass objects to the component states during transitions, so that might be an alternative. If I've researched correctly then it should be something like:

```ts
// Route
{
      name: 'route',
      url: '/my-page',
      component: MyComponent,
      params: {
        source: null  // default value
      }
    }

// Transition
stateSerivce.go('route', { source: this.dataModel });   // Send object as param to state

// Access
const source = this.uiRouterGlobals.params.source;
```

Reply:
i'm not sure either way is the best way. It was just a way to try and cut a call out but it doesn't work on the refresh i'll remove it